### PR TITLE
[Bugfix] Fix a bug when 0 dim linalg det backwards

### PR DESCRIPTION
**Pull request by Skylion007**
Zero strided tensor should have gradients be a zero stride tensor, not an undefined Tensor (implicitly 1). This can cause issues with 2nd order and higher derivatives. It also violates the invariants that grad.shape == tensor.shape

### CODE REVIEW

File: test/test_linalg.py [hunk 1]
```diff
@@ -1050,6 +1050,16 @@ def test_det_backward(self, device, dtype):
...
+        # When A has 0 elements (e.g. empty batch), backward should return a
+        # zeros tensor with the same shape as A, not an undefined tensor.
+        for shape in [(0, 3, 3), (2, 0, 0)]:
+            A = torch.randn(shape, device=device, dtype=dtype, requires_grad=True)
+            det = torch.linalg.det(A)
+            det.backward(torch.ones_like(det))
+            self.assertIsNotNone(A.grad)
+            self.assertEqual(A.grad.shape, A.shape)
+            self.assertEqual(A.grad, torch.zeros_like(A))
+
```
Comment (lezcano) [inline test/test_linalg.py:1062]: This is not a regression test tho, is it?

**Review by lezcano — APPROVED** [review]
SG in principle, but should add a proper regression test.

File: test/test_linalg.py (hunk 1 above)
Comment (lezcano) [reply test/test_linalg.py:1062]: ah, you are checking the shape, fine.
